code/game/objects/items/storage/backpack.dm

```diff
@@ -706,6 +706,18 @@
 		to_chat(user, span_notice("You refill [FT] with [lowertext(FT.caliber)]."))
 		FT.update_icon()
 
+	else if(istype(I, /obj/item/weapon/twohanded/rocketsledge))
+		var/obj/item/weapon/twohanded/rocketsledge/RS = I
+		if(RS.reagents.get_reagent_amount(/datum/reagent/fuel) == RS.max_fuel || !reagents.total_volume)
+			return ..()
+
+		var/fuel_transfer_amount = min(reagents.total_volume, (RS.max_fuel - RS.reagents.get_reagent_amount(/datum/reagent/fuel)))
+		reagents.remove_reagent(/datum/reagent/fuel, fuel_transfer_amount)
+		RS.reagents.add_reagent(/datum/reagent/fuel, fuel_transfer_amount)
+		playsound(loc, 'sound/effects/refill.ogg', 25, 1, 3)
+		to_chat(user, span_notice("You refill [RS] with fuel."))
+		RS.update_icon_state()
```

Comment:
never call directly, call update_icon() (which in turns calls update_icon_state, with some signals and overlay changes as well)